pkg/pillar/cmd/zedagent/handlecipherconfig.go

```diff
@@ -38,9 +47,10 @@ func parseCipherContext(ctx *getconfigContext,
 
 	cipherCtxHash = newHash
 
+	invalidateCipherContextDependenciesList()
```

Comment:
Do we really need to have the declaration of this function and an actual call here in the cipher parse procedure? What if invalidate those hashes in the parseConfig() just before the whole parsing starts ? 

Also I see that this invalidation happens unconditionally: parseCipherContext() will be always called before parseNetworkXObjectConfig() or parseInstanceConfig() or parseDataStoreConfig(), so are we correct in assumption that these 'previous' hashes are really used?

Reply:
I put it here because we want to invalidate only on CipherContexts's changes. And we check were they changed by comparing the hashes few lines above `bytes.Equal(newHash, cipherCtxHash)`. So it is conditionally: if CipherContexts's are the same we will not invalidate.

> are we correct in assumption that these 'previous' hashes are really used?

Sorry, probably I do not understand the question. By setting 'previous' hash to an empty slice we will force all dependent parsers to re-parse and populate updates as they may use CipherContext.

Reply:
> I put it here because we want to invalidate only on CipherContexts's changes. And we check were they changed by comparing the hashes few lines above `bytes.Equal(newHash, cipherCtxHash)`. So it is conditionally: if CipherContexts's are the same we will not invalidate.


My bad, missed that:

	if bytes.Equal(newHash, cipherCtxHash) {
		return
	}